tests/test-simple

```diff
@@ -89,10 +89,11 @@ test_simple_hg()
     printf '\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0' > .hg/dirstate
     printf '\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0' >> .hg/dirstate
     assert_vcprompt "hg initial nodeid" "hg:" "%n:%r"
-
+    _ping
     mkdir subdir
     cd subdir
     assert_vcprompt "hg subdir" "foo"
```

Comment:
Test should fail in between `_ping` and `_pong` due to an invalid free somewhere within `vcprompt`, but the tests don't even notice it.  I'd check to make sure that the invalid free is present already (in current master), but either way this shows that the tests don't seem to work reliably.